### PR TITLE
Use .tar.zst instead of .tar.xz

### LINKED ISSUE #6: Don't hardcore PKG_EXT = '.tar.xz'
Whenever we switch to .tar.zstd this will break.

### PR DESCRIPTION
**Pull request by kylelaker**
Fixes #6

### CODE REVIEW

**Comment by kylelaker** [comment]
If this needs to be completely agnostic of PKG_EXT, that can be done too. I just started out with the most similar solution to the current

**Comment by coderobe** [comment]
It does need to be agnostic, to be both backward and forward-compatible, for the transition period anyways.

File: archive-cleaner [hunk 1]
```diff
@@ -68,7 +68,7 @@ def get_packages_from_buildinfo(archive, filename, archive_entry, arch):
     packages = []
     if buildinfo_format > 0:
         for package in installed:
-            pkgfilename = f'{package}.pkg.tar.xz'
+            pkgfilename = f'{package}.pkg.{PKG_EXT}'
```
Comment (0x9fff00) [inline archive-cleaner]: `.{PKG_EXT}` will become `..tar.zst`

File: archive-cleaner [hunk 2]
```diff
@@ -81,8 +81,8 @@ def get_packages_from_buildinfo(archive, filename, archive_entry, arch):
             packages.append(pkgfilename)
         return packages
     for package in installed:
-        filename_any = f'{package}-any.pkg.tar.xz'
-        filename_arch = f'{package}-{arch}.pkg.tar.xz'
+        filename_any = f'{package}-any.pkg.{PKG_EXT}'
+        filename_arch = f'{package}-{arch}.pkg.{PKG_EXT}'
```
Comment (0x9fff00) [inline archive-cleaner]: Same as above

**Comment by coderobe** [comment]
I've implemented a more advanced approach for archive.sh in https://github.com/archlinux/archivetools/pull/8 similar to the one deployed in our other projects - as of now, the patch for archive-cleaner is still missing from that, though.

**Comment by anthraxx** [comment]
That regex looks not strict enough, the dots are interpreted as any chars, aren't they?
Also the second part should not consist of two optional engages, one dot and one suffix but a combined one, a single optional range that consists of dot+suffix otherwise you match on pkg.targz as well

**Comment by coderobe** [comment]
Good point, will fix. Also you replied on the wrong thread :P

**Comment by anthraxx** [comment]
#9 is in process fixing the named issues

**Pull request closed without merging**